staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -536,6 +539,18 @@ func (f *DeltaFIFO) Pop(process PopProcessFunc) (interface{}, error) {
 			continue
 		}
 		delete(f.items, id)
+		// Only log traces if the queue depth is greater than 10 and it takes more than
+		// 1 second to process one item from the queue.
```

Comment:
I've made up the number based on local experimentation, sending 300 events to the queue and adding a 1 second delay on one of the handlers, it never went higher than ~30 (but this is totally random)